src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -320,8 +321,19 @@ public void init(boolean forceNewConnection) {
         clearInstances(instances);
         clearInstances(brokenInstances);
 
+
         Reporter reporter = appConfig.getReporter();
 
+        String action = appConfig.getAction();
+
+        if( action.equals( AppConfig.ACTION_LIST_JVMS )) {
+            List<VirtualMachineDescriptor> descriptors = com.sun.tools.attach.VirtualMachine.list();
+            for( VirtualMachineDescriptor descriptor : descriptors ) {
+                System.out.println( "\tJVM id " + descriptor.id() + ": '" + descriptor.displayName() + "'" );
+            }
+            return;
+        }
+
```

Comment:
I think this whole block should be moved to the `main()` method instead, i.e. [here](https://github.com/DataDog/jmxfetch/blob/master/src/main/java/org/datadog/jmxfetch/App.java#L83) for instance.